doc/docs/modules/ROOT/pages/gds.adoc

```diff
@@ -18,11 +18,222 @@ All of the link:{url-neo4j-gds-manual}[functionality of GDS^] is used by issuing
 accessible via Spark, because the Neo4j Connector for Apache Spark can issue Cypher queries and read their results back.  This combination means
 that you can use Neo4j and GDS as a graph co-processor in an existing ML workflow that you may implement in Apache Spark.
 
-== Example
+== GDS first-class support
+
+Neo4j Spark Connector added first-class support to Graph Data Science library let's see how it works.
+
+=== Limitations
+
+We don't support `mutate` or `write` procedure as the don't return any kind of usable information in the dataframe, you can achieve the same by joining dataframes and then
+use the Neo4j Spark connector to write back the data to whatever Neo4j instance you want.
+
+=== Example
+
+Immagine that we want to replicate the example for the Page Rank algorithm
+detailed link:{url-neo4j-gds-manual}algorithms/page-rank/#algorithms-page-rank-examples[here in the GDS manual^].
+The related Spark example looks like the following (we'll assume that the data is already in Neo4j).
+
+1. Create the projection graph: the following Python code will project a graph using a native projection and store it in the graph catalog under the name 'myGraph'.
+
+[source,python]
+----
+    # we'll assume that `spark` variable is already present
+    spark.read.format("org.neo4j.spark.DataSource") \
+      .option("url", "neo4j://localhost:7687") \
+      .option("gds", "gds.graph.project") \
+      .option("gds.graphName", "myGraph") \
+      .option("gds.nodeProjection", "Page") \
+      .option("gds.relationshipProjection", "LINKS") \
```

Comment:
Can you pass complex map values to the projection keys?

Reply:
Complex maps are handled via dot notation as explained here:
https://github.com/neo4j-contrib/neo4j-spark-connector/pull/506/files#diff-79ddc5872caa72c3a3869bf25a1670a345b513fb1a9876d80831d09a40893156R163-R164

We have a unit test for the implementation that handles the feature, can you please check it?
https://github.com/neo4j-contrib/neo4j-spark-connector/pull/506/files#diff-27238ebeb4d9b3fcce40df973d7dc14377ef6c4f79b96b82849b0705dcc5d8beR15-R51

Does it cover enough? Can we improve it?